### PR TITLE
[Do not merge] Feature c++ client

### PR DESCRIPTION
**Pull request by oleorhagen**

### CODE REVIEW

**Comment by lluiscampos** [comment]
@oleorhagen I assume this is going to live (for now) on a `feature-c++-client` branch, right? Then please create this branch as per today's master, and launch the PR against that. Otherwise we risk clicking :green_circle: and :boom:

File: mender-deb-package [hunk 1]
```diff
@@ -128,6 +129,9 @@ prepare_recipe() {
     if [ "${debian_recipe}" = "debian-2.1.x" ]; then
       # Skip for 2.1.x, special handling in debian/rules
       true
+    elif [ "${debian_recipe}" = "debian-master" ]; then
+      # skip
+      true
```
Comment (lluiscampos) [inline mender-deb-package:134]: Okey for the feature branch, but we will need to revert at some point.

File: Dockerfile [hunk 2]
```diff
@@ -5,9 +5,15 @@ FROM $DISTRO:$VERSION
 RUN export DEBIAN_FRONTEND=noninteractive && \
     apt-get update && \
     apt-get install -y \
-    build-essential dh-make \
-    git wget curl \
-    debhelper devscripts
+    build-essential \
+    crossbuild-essential-armhf  \
```
[the remark this answers is not in the document]
Comment (lluiscampos) [reply Dockerfile:9]: As we are using the official toolchain, I would then clean up all the special handling for armhf packages below. I am refereeing to the layers starting with `RUN if [ "${DISTRO}" = "debian" -a "${ARCH}" = "armhf" ]; then \ ...`

File: mender-deb-package [hunk 3]
```diff
@@ -189,13 +193,50 @@ build_packages() {
...
   fi
 
+  # Install the builder architecture independent dependencies from the debian/control file
+  mk-build-deps --install --build-indep --tool='apt-get -o Debug::pkgProblemResolver=yes --no-install-recommends --yes' debian/control
+
+  # HOST - the machine arch we are building _for_
+  HOST_ARCH=${ARCH}
+
+  dpkg --add-architecture ${ARCH}
+  apt update
```
Comment (lluiscampos) [inline mender-deb-package:226]: This we are doing already in the Dockerfile (except for the armhf on Debian special handling which I think we should remove in this PR)

File: recipes/mender-client/debian-master/rules [hunk 4]
```diff
@@ -1,42 +1,21 @@
...
-	done
-
-	install -d -m 755 debian/mender-client/etc/mender/inventory
-	mv debian/mender-client/usr/share/mender/inventory/* debian/mender-client/etc/mender/inventory/
-	for file in debian/mender-client/etc/mender/inventory/*; do \
-		ln -sf /etc/mender/inventory/"$$(basename "$$file")" debian/mender-client/usr/share/mender/inventory/; \
-	done
+#override_dh_auto_configure:
+#dh_auto_configure -- \
+#-D BOOST_ALL_DYN_LINK=ON
```
Comment (lluiscampos) [inline recipes/mender-client/debian-master/rules:18]: Everything commented out here, or?

File: mender-deb-package [hunk 5]
```diff
@@ -189,13 +193,50 @@ build_packages() {
...
+    # We want the patches applied for Debian Buster
+    rm -f debian/source/local-options
+    skip_patches_flag=
+  else
+    # Skip applying the patches
+    rm -f debian/patches/series
   fi
 
+  # Install the builder architecture independent dependencies from the debian/control file
+  mk-build-deps --install --build-indep --tool='apt-get -o Debug::pkgProblemResolver=yes --no-install-recommends --yes' debian/control
```
[the remark this answers is not in the document]
Comment (oleorhagen) [reply mender-deb-package:220]: It builds a `package`, which is basically the `dev` package you are talking about.

[the remark this answers is not in the document]
Comment (oleorhagen) [reply mender-deb-package:220]: Why they do it this way, I dunno

**Pull request closed without merging**